diesel/src/query_builder/ast_pass.rs

```diff
@@ -240,6 +253,22 @@ where
     pub fn backend(&self) -> &DB {
         self.backend
     }
+
+    #[cfg_attr(
+        not(feature = "i-implement-a-third-party-backend-and-opt-into-breaking-changes"),
+        doc(hidden)
+    )] // This is used by the `sql_function` macro
+    #[cfg_attr(
+        doc_cfg,
+        doc(cfg(feature = "i-implement-a-third-party-backend-and-opt-into-breaking-changes"))
+    )]
+    pub fn should_skip_from(&self) -> bool {
```

Comment:
Would we prefer to expose the whole options struct instead?
Is the name too vague?